pkg/api/defaults-kubelet.go

```diff
@@ -92,7 +92,6 @@ func (cs *ContainerService) setKubeletConfig() {
 
 	// If no user-configurable kubelet config values exists, use the defaults
 	setMissingKubeletValues(o.KubernetesConfig, defaultKubeletConfig)
-	addDefaultFeatureGates(o.KubernetesConfig.KubeletConfig, o.OrchestratorVersion, "", "")
```

Comment:
The primary motivation for this PR was observing this weird thing in the default kubelet assignments flow.

Reply:
so is the thinking that this was unnecessary? any chance this was meant to be a guard for backcompat upgrade or anything like that?

Reply:
As I read the functional implementation of `addDefaultFeatureGates`, it does nothing. That func just adds to the pre-existing `--feature-gates` string. So passing in `""` does... nothing.

Also, the fact that the next statement actually does add stuff is additionally suspicious.

I was a little confused too, which is why I wrote the tests to prove that we weren't doing anything. Feel free to look through the tests and note any user scenarios that should be added. We can then do a before/after to prove that the tests both with and without the above line (to prove no-op).

Reply:
👍